huaweicloud/services/acceptance/rf/resource_huaweicloud_rf_stack_test.go

```diff
@@ -0,0 +1,292 @@
+  name        = "%[1]s"
+  description = "Create by acc test"
+}
+`, name)
+}
+
+func testAccStack_basic_step2(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_rf_stack" "test" {
+  name        = "%[1]s"
+  description = "Create by acc test"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_body = <<EOT
+{
+  "terraform": {
+    "required_providers": [
+      {
+        "huaweicloud": {
+          "source": "huawei.com/provider/huaweicloud",
+          "version": ">= 1.41.0"
+        }
+      }
+    ]
+  },
+  "provider": {
+    "huaweicloud": {
+      "region": "%[2]s"
+    }
+  },
+  "resource": {
+    "huaweicloud_vpc": {
+      "test": {
+        "name": "%[1]s",
+        "cidr": "192.168.0.0/16"
+      }
+    }
+  }
+}
+EOT
+}
+`, name, acceptance.HW_REGION_NAME)
+}
+
+func testAccStack_basic_step3(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_rf_stack" "test" {
+  name        = "%[1]s"
+  description = "Create by acc test"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_body = <<EOT
+{
+  "terraform": {
+    "required_providers": [
+      {
+        "huaweicloud": {
+          "source": "huawei.com/provider/huaweicloud",
+          "version": ">= 1.41.0"
+        }
+      }
+    ]
+  },
+  "provider": {
+    "huaweicloud": {
+      "region": "%[2]s"
+    }
+  },
+  "resource": {
+    "huaweicloud_vpc": {
+      "test": {
+        "name": "%[1]s",
+        "cidr": "192.168.0.0/16"
+      }
+    },
+    "huaweicloud_vpc_subnet": {
+      "test": {
+        "vpc_id": "$${huaweicloud_vpc.test.id}",
+        "name": "$${var.subnet_name}",
+        "cidr": "$${cidrsubnet(huaweicloud_vpc.test.cidr, 4, 1)}",
+        "gateway_ip": "$${cidrhost(cidrsubnet(huaweicloud_vpc.test.cidr, 4, 1), 1)}"
+      }
+    }
+  },
+  "variable": {
+    "subnet_name": {
+      "type": "string"
+    }
+  }
+}
+EOT
+
+  vars_body = <<EOT
+subnet_name = "%[1]s"
+EOT
+}
+`, name, acceptance.HW_REGION_NAME)
+}
+
+func testAccStack_uri_step1(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_rf_stack" "test" {
+  name = "%[1]s"
+}
+`, name)
+}
+
+func testAccStack_uri_step2(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_rf_stack" "test" {
+  name = "%[1]s"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_uri = "%[2]s"
+}
+`, name, acceptance.HW_RF_TEMPLATE_ARCHIVE_NO_VARS_URI)
+}
+
+func testAccStack_uri_step3(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_rf_stack" "test" {
+  name = "%[1]s"
+
+  agency {
+    name          = "rf_admin_trust" // System RF agency
+    provider_name = "huaweicloud"
+  }
+
+  template_uri = "%[2]s"
+  vars_uri     = "%[3]s"
+}
+`, name, acceptance.HW_RF_TEMPLATE_ARCHIVE_URI, acceptance.HW_RF_VARIABLES_ARCHIVE_URI)
```

Comment:
pls create an obs in the config instead

Reply:
the template needs ZIP file. the obs resource does not support this function.